internal/api/group.go

```diff
@@ -21,6 +21,54 @@ func initGroup(apiRouter *mux.Router, context *Context) {
 	groupRouter.Handle("", addContext(handleGetGroup)).Methods("GET")
 	groupRouter.Handle("", addContext(handleUpdateGroup)).Methods("PUT")
 	groupRouter.Handle("", addContext(handleDeleteGroup)).Methods("DELETE")
+
+	groupRouter.Handle("/mattermost/{version:[A-Za-z0-9.]+}", addContext(handleChangeMattermostVersion)).Methods("PUT")
+}
+
+// handleChangeMattermostVersion responds to PUT
+// /api/group/{group}/mattermost/{version}, upgrading or downgrading
+// the installation to the Mattermost version embedded in the request.
+func handleChangeMattermostVersion(c *Context, w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+
+	groupID, ok := vars["group"]
+	if !ok {
+		c.Logger.Error("couldn't get group ID from request parameters")
```

Comment:
nit: I would prefer our errors start with `failed to` like the most of the other API error messages, but not a big deal

Reply:
ah, sorry, you left a comment like this on a previous PR and I didn't understand the reason for the wording change so it didn't "stick" with me; uniform language is important and I will fix this before merge.